packages/form-core/src/choice-group/ChoiceGroupMixin.js

```diff
@@ -50,13 +57,62 @@ export const ChoiceGroupMixin = dedupeMixin(
       }
 
       set serializedValue(value) {
-        this._setCheckedElements(value, (el, val) => el.serializedValue.value === val);
+        if (this.__isInitialSerializedValue) {
+          this.__isInitialSerializedValue = false;
+          this.registrationComplete.then(() => {
+            this._setCheckedElements(value, (el, val) => el.serializedValue.value === val);
+          });
+        } else {
+          this._setCheckedElements(value, (el, val) => el.serializedValue.value === val);
+        }
       }
 
       constructor() {
         super();
         this.multipleChoice = false;
         this._repropagationRole = 'choice-group'; // configures event propagation logic of FormControlMixin
+
+        this.__isInitialModelValue = true;
+        this.registrationComplete = new Promise((resolve, reject) => {
+          this.__resolveRegistrationComplete = resolve;
+          this.__rejectRegistrationComplete = reject;
+        });
+        this.registrationComplete.done = false;
+        this.registrationComplete.then(
+          () => {
+            this.registrationComplete.done = true;
```

Comment:
I though that was poorly supported... but it's actually pretty good... 
will need to check if it's in the polyfill... but I'm not sure if it would improve the code a lot 😅

also could be done later if needed?

Reply:
Which polyfill, the webcomponents one? Definitely makes the code a bit more readable for me, but not a blocker.

Reply:
I suppose he is talking about finally. It’s not supported by IE.

Reply:
It's in core js but not every consumers uses it, or loads it on every browser. So it's a bit of a gray area :( Ironically async await is compiled away, so using finally with that isn't a problem.